src/react/hooks/__tests__/useMutation.test.tsx

```diff
@@ -268,7 +268,7 @@ describe('useMutation Hook', () => {
   });
 
   describe('Optimistic response', () => {
-    it('should support optimistic response handling', async () => {
+    it('should support optimistic response handling', async () => new Promise((resolve, reject) => {
```

Comment:
I would love to shorten this to
```suggestion
    it.async('should support optimistic response handling', async (resolve, reject) => {
```
I will push a follow-up commit to this PR if I can figure out a good way to add that `it.async` API.